metadata-ingestion/src/datahub/ingestion/source/vertica/common.py

```diff
@@ -0,0 +1,1628 @@
+            self.models_scanned += 1
+        elif ent_type == "OAuth":
+            self.oauth_scanned += 1
+        else:
+            super().report_entity_scanned(name, ent_type)
+
+#Extended BasicSQLAlchemyConfig to config for projections,models and oauth metadata.
+class SQLAlchemyConfigVertica(BasicSQLAlchemyConfig):
+
+    projection_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for projection to filter in ingestion. Specify regex to match the entire projection name in database.schema.projection format. e.g. to match all tables starting with customer in Customer database and public schema, use the regex 'Customer.public.customer.*'",
+    )
+    models_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for ml models to filter in ingestion. "
+    )
+    oauth_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for OAuth to filter in ingestion. "
+    )
+
+    include_projections: Optional[bool] = Field(
+        default=True, description="Whether projections should be ingested."
+    )
+    include_models: Optional[bool] = Field(
+        default=True, description="Whether Models should be ingested."
+    )
+    include_oauth: Optional[bool] = Field(
+        default=True, description="Whether OAuth should be ingested."
+    )
+    include_view_lineage: Optional[bool] = Field(
+        default=True, description="Whether lineages should be ingested for views"
+    )
+    include_projection_lineage: Optional[bool] = Field(
+        default=True, description="Whether lineages should be ingested for Projection"
+    )
+
+
+# config flags to emit telemetry for
+config_options_to_report = [
+    "include_views",
+    "include_tables",
+    "include_projections",
+    "include_models",
+    "include_oauth",
+    "include_view_lineage",
+    "include_projection_lineage"
+
+]
+
+# flags to emit telemetry for
+profiling_flags_to_report = [
+    "turn_off_expensive_profiling_metrics",
+    "profile_table_level_only",
+    "include_field_null_count",
+    "include_field_min_value",
+    "include_field_max_value",
+    "include_field_mean_value",
+    "include_field_median_value",
+    "include_field_stddev_value",
+    "include_field_quantiles",
+    "include_field_distinct_value_frequencies",
+    "include_field_histogram",
+    "include_field_sample_values",
+    "query_combiner_enabled",
+]
+
+
+class SchemaKeyHelper(SchemaKey):
+    numberOfProjection: Optional[str]
+    udxsFunctions : Optional[str] = None
+    UDXsLanguage : Optional[str] = None
+
+
+class DatabaseKeyHelper(DatabaseKey):
+    clusterType : Optional[str] = None
+    clusterSize : Optional[str] = None
+    subClusters : Optional[str] = None
+    communalStoragePath : Optional[str] = None
+
+
+class VerticaSQLAlchemySource(SQLAlchemySource):
```

Comment:
it still looks like a large portion of this file was copied from SQLAlchemySource.

only the methods that are changed should be overridden (with comments about exactly what changed), and you should be calling super().<method> wherever possible to reuse the existing logic

Reply:
thanks for the review, we cleaned some functions, and now only functions that have code changes have been reused, we added integrations testing for Vertica and all unit tests have been moved to dialect since it contains most of the logic. Please check the latest commit for changes.